qa/tasks/cephfs/test_volumes.py

```diff
@@ -4185,6 +4185,62 @@ def test_subvolume_user_metadata_list_and_remove_for_legacy_subvolume(self):
         # verify trash dir is clean.
         self._wait_for_trash_empty()
 
+    def test_subvolume_rm_when_cluster_is_full(self):
```

Comment:
You could write a workunit test and add it to qa/suites/fs/full suite. I think the fs:full suite already has a test case for suvolume rm when osd is full scenario (./qa/workunits/fs/full/subvolume_rm.sh)

I think you need to write a similar test with retained-snapshot subvolume deletion

I think it's complicated to write a osd full test here in this suite as this runs on cluster with multiple osds. You need to get the osd full calculation right and fill the data accordingly. In fs:full suite it's single node single osd cluster, it's easy to write a test there.

Reply:
@neesingh-rh ping?

Reply:
Test has already been added. forget to ack this comment